test/weights.jl

```diff
@@ -3,355 +3,370 @@ using Base.Test
 using Compat
 import Compat: view
 
-@test isa(weights([1, 2, 3]), WeightVec{Int})
-@test isa(weights([1., 2., 3.]), WeightVec{Float64})
-@test isa(weights([1 2 3; 4 5 6]), WeightVec{Int})
-
-@test isa(WeightVec([1, 2, 3], 6), WeightVec{Int})
-
-@test isempty(weights(Float64[]))
-@test size(weights([1, 2, 3])) == (3,)
-
-w  = [1., 2., 3.]
-wv = weights(w)
-@test eltype(wv) === Float64
-@test length(wv) === 3
-@test values(wv) === w
-@test sum(wv) === 6.0
-@test !isempty(wv)
-
-b  = trues(3)
-bv = weights(b)
-@test eltype(bv) === Bool
-@test length(bv) === 3
-@test values(bv) === b
-@test sum(bv)    === 3
-@test !isempty(bv)
-
-ba = BitArray([true, false, true])
-sa = sparsevec([1., 0., 2.])
-
-@test sum(ba, wv) === 4.0
-@test sum(sa, wv) === 7.0
-
-## wsum
-
-x = [6., 8., 9.]
-w = [2., 3., 4.]
-p = [1. 2. ; 3. 4.]
-q = [1., 2., 3., 4.]
-
-@test wsum(Float64[], Float64[]) === 0.0
-@test wsum(x, w) === 72.0
-@test wsum(p, q) === 29.0
-
-## wsum along dimensions
+@testset "StatsBase.Weights" begin
+weight_funcs = (aweights, fweights, pweights)
+
+@testset "Construction" begin
```

Comment:
@nalimilan Is it alright if I keep the `@testset`s when they're logically tied to my new tests or the ones wrapping the entire file (they aren't required, but should change the diff much and will keep the test output clean)?